src/scripts/api.ts

```diff
@@ -173,141 +154,180 @@ export const createTableFromSelections = async (selectedOptions: IStateOptions)
-        "AREA OPERATED: (1,000 TO 5,000 ACRES)": {
-          "AREA OPERATED: (1,000 TO 1,999 ACRES)": 0,
-          "AREA OPERATED: (2,000 TO 4,999 ACRES)": 0
-        },
-        "AREA OPERATED: (5,000 OR MORE ACRES)": {
-          "AREA OPERATED: (5,000 OR MORE ACRES)": 0
-        }
-      };
-      const totalKeys = Object.keys(acreTotals);
-      for (const total of totalKeys) {
-        const subTotalKeys = Object.keys(acreTotals[total]);
-        const dataItems = data.filter((dataItem: any) => subTotalKeys.includes(dataItem.domaincat_desc));
-        dataItems.forEach((dataItem: any) => {
-          acreTotals[total][dataItem.domaincat_desc] = dataItem.Value.replace(/,/g, "");
-        });
-        const codapColumnName = attrToCODAPColumnName[total].attributeNameInCodapTable;
-        // sum up all the values of acreTotals[total]
-        const onlyNumbers = subTotalKeys.map((key) => Number(acreTotals[total][key]));
-        const sum = onlyNumbers.reduce((acc, cur) => acc + cur);
-        values[codapColumnName] = sum;
-      }
+
+  if (attribute === "Total Farmers" && (years.length > 1 && years.includes("2017"))) {
+    // we need to make two requests -- one for the total farmers in 2017, and one for the total farmers in all other years
+    const req2017 = createRequest({...reqParams, years: ["2017"]});
+    const reqOtherYears = createRequest({...reqParams, years: years.filter((year) => year !== "2017")});
+    const res2017 = await fetchDataWithRetry(req2017, setReqCount);
+    const resOtherYears = await fetchDataWithRetry(reqOtherYears, setReqCount);
+    if (res2017 && resOtherYears) {
+      return [...res2017.data, ...resOtherYears.data];
     } else {
-      data.forEach((dataItem: any) => {
-        const dataItemDesc = attribute === "Economic Class" ? dataItem.domaincat_desc : dataItem.short_desc;
-        const codapColumnName = attrToCODAPColumnName[dataItemDesc].attributeNameInCodapTable;
-        values[codapColumnName] = dataItem.Value;
-      });
+      console.log(`Error: did not receive resposnse for this request:`, req);
     }
   } else {
-    // eslint-disable-next-line no-console
-    console.log(`Error: did not receive response for this request:`, req);
+    const res = await fetchDataWithRetry(req, setReqCount);
+    if (res) {
+      return res.data;
+    } else {
+      // eslint-disable-next-line no-console
+      console.log(`Error: did not receive response for this request:`, req);
+    }
   }
-  return values;
+};
+
+export const fetchDataWithRetry = async (req: string, setReqCount: ISetReqCount, maxRetries = 3,) => {
+  let retries = 0;
+  while (retries < maxRetries) {
+    try {
+      const response = await fetchJsonp(req, { timeout: 30000 }); // Increase the timeout
+      const json = await response.json();
+      setReqCount((prevState) => {
+        const completed = prevState.completed + 1 > prevState.total ? prevState.total : prevState.completed + 1;
+        return {...prevState, completed};
```

Comment:
No need to change the code but line 322 could be folded into line 323 like this:

`return {...prevState, completed: Math.min(prevState.total, prevState.completed + 1)};`